connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -79,11 +83,30 @@ public ConnectorsResource(Herder herder, WorkerConfig config) {
 
     @GET
     @Path("/")
-    public Collection<String> listConnectors(final @QueryParam("forward") Boolean forward) throws Throwable {
-        FutureCallback<Collection<String>> cb = new FutureCallback<>();
-        herder.connectors(cb);
-        return completeOrForwardRequest(cb, "/connectors", "GET", null, new TypeReference<Collection<String>>() {
-        }, forward);
```

Comment:
Aren't we losing the ability to forward the request to the leader?

Reply:
i believe this is not needed..

if you look at something like: https://github.com/apache/kafka/blob/7d9b22e35efd1293b6e639cdbed5abb0c1140d30/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java#L154 it seems like we don't need to forward (at least) some of the requests.

and specifically for `/connectors`, it looks like the herder in each worker ends up with all the necessary info when i test this locally.

Reply:
I was mostly asking to get justification why this change is ok.

Previously, the call to `herder.connectorStatus(connector)` might not complete if the herder is expecting a rebalance, at which point the request would have been forwarded to the leader. The `completeOrForwardRequest` submits to the herder's single thread the logic of checking the response and potentially forwarding. It's important that this be done for mutating methods, but this is a read-only method. The only time a local implementation might have problems with listing connectors is if the local herder knows its config store has fallen behind and hasn't been able to keep up with reading the latest changes. But interestingly, the distributed herder always returns the list of connectors from its in-memory cache, and never consults the store anyway.

With the new code, the `herder.connector()` and `herder.connectorInfo(String)` methods always return representations that are from the local herder's config store, and the `herder.connectorStatus()` method (which existed previously) also only looked in the local herder's status store. 

If this is true, then we should not need to forward. It will mean, however, that if there is a network partition that isolates a worker from the Kafka cluster, you will still get a response based upon the local worker's config and status stores, but it might be stale and may not match the same result returned from a different worker. But it looks like the old version of this method did that anyway, since DistributedHerder always returns the names of the connectors from it's in-memory cache.

In the end, I agree that I see no reason to forward individual requests *for listing the connectors* to the leader.